websites/S/SiriusXM/presence.ts

```diff
@@ -1,86 +1,116 @@
+	startTimestamp: number | null = null;
+
+const getTextContent = (selector: string): string =>
+		document.querySelector(selector)?.textContent ?? "",
+	getImageSrc = (selector: string): string =>
+		document.querySelector(selector)?.querySelector("img")?.src ?? sxmLogo,
+	pathDetailsMap: { [key: string]: string } = {
+		"/player/home/for-you": "Viewing SiriusXM Home",
+		"/player/home": "Viewing SiriusXM Home",
+		"/player/home/music": "Viewing Music Home",
+		"/player/home/sports": "Viewing Sports Home",
+		"/player/home/talk-and-podcast": "Viewing Talk & Podcasts Home",
+		"/player/my-library": "Viewing Their Library",
+		"/player/search": "Searching SiriusXM...",
+		"/player/channel-linear": "Viewing Channel:",
+		"/player/show": "Viewing Show:",
+		"/player/talent": "Viewing Host:",
+		"/player/episode-video": "Viewing Video Episode: ",
+		"/player/episode-audio": "Viewing Audio Episode: ",
+		"/player/genre": "Viewing Genre:",
+		"/player/event": "Viewing Event:",
+		"/player/settings": "Viewing Their Settings",
+		"/player/welcome": "Viewing The Welcome Page",
+		"/player/login": "Logging in...",
 	};
 
-	switch (document.location.pathname) {
-		case "/home/foryou": {
-			presenceData.details = "Viewing SiriusXM Home";
-			break;
-		}
-		case "/home/music": {
-			presenceData.details = "Viewing Music Home";
-			break;
-		}
-		case "/home/sports": {
-			presenceData.details = "Viewing Sports Home";
-			break;
-		}
-		case "/home/news": {
-			presenceData.details = "Viewing News Home";
-			break;
-		}
-		case "/home/entertainment": {
-			presenceData.details = "Viewing Talk Home";
-			break;
-		}
-		case "/home/howard": {
-			presenceData.details = "Viewing Howard Stern Home";
-			break;
-		}
-		case "/favorites/channels": {
-			presenceData.details = "Viewing Favorite Channels";
-			break;
-		}
-		case "/favorites/shows": {
-			presenceData.details = "Viewing Favorite Shows";
-			break;
-		}
-		case "/favorites/episodes": {
-			presenceData.details = "Viewing Favorite Episodes";
-			break;
-		}
-		case "/recently-played": {
-			presenceData.details = "Viewing Recently Played Stations";
-			break;
-		}
-		case "/query": {
-			presenceData.details = "Searching SiriusXM";
-			break;
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = { largeImageKey: sxmLogo, startTimestamp },
+		title = getTextContent(
+			"#root > div > div._BaseLayout_711jo_11 > main > div > div > header > div > div > div._EntityHeaderMetadata_1mpio_27 > div._EntityHeaderTexts_1mpio_32 > div._EntityHeaderTitle_1mpio_43 > h1"
+		),
+		subtitle = getTextContent(
+			"#root > div > div._BaseLayout_711jo_11 > main > div > div > header > div > div > div._EntityHeaderMetadata_1mpio_27 > div._EntityHeaderTexts_1mpio_32 > div._EntityHeaderSubtitle_1mpio_47"
+		),
+		channelImage = getImageSrc(
+			"#root > div > div._BaseLayout_711jo_11 > main > div > div > header > div > div > div._EntityImageContainer_1mpio_65 > span > span > span > span"
+		),
```

Comment:
Ideally, use less specific selectors and avoid selectors including computer-generated class names. A useful tool is the [`*=`](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attrvalue_6) selector